cleanlab/segmentation/rank.py

```diff
@@ -0,0 +1,240 @@
+    pred_probs: np.ndarray,
+    method: str = "softmin",
+    batch_size: int = 10000,
+    n_jobs: Optional[int] = 1,
+    verbose: bool = True,
+    **kwargs
+) -> Tuple[np.ndarray,np.ndarray]:
+    """Returns a label quality score for each image.
+
+    This is a function to compute label quality scores for standard (multi-class) classification datasets,
+    where lower scores indicate labels less likely to be correct.
+
+    Score is between 0 and 1.
+
+    1 - clean label (given label is likely correct).
+    0 - dirty label (given label is likely incorrect).
+
+    Parameters
+    ----------
+    labels : np.ndarray 
+      A discrete array of noisy labels for a classification dataset, i.e. some labels may be erroneous.
+      *Format requirements*: for dataset with K classes, each pixel must be integer in 0, 1, ..., K-1.
+      For a standard (multi-class) classification dataset where each example is labeled with one class,
+      `labels` should be 3-D array of shape ``(N,H,W,)``. 
+      
+    Tip: If your labels are one hot encoded you can `np.argmax(labels_one_hot,axis=1)` assuming that `labels_one_hot` is of dimension (N,K,H,W)
+    before entering in the function
+
+    pred_probs : np.ndarray
+      An array of shape ``(N,K,H,W,)`` of model-predicted class probabilities,
+      ``P(label=k|x)``. Each pixel contains an array of K classes, where for 
+      an example `x` the array at each pixel contains the model-predicted probabilities 
+      that `x` belongs to each of the K classes.
+      
+    method : {"softmin", "num_pixel_issues"}, default="softmin"
+      Label quality scoring method.
+
+    batch_size : int, optional
+      For num_pixel_issues:
+      Size of mini-batches to use for estimating the label issues.
+      To maximize efficiency, try to use the largest `batch_size` your memory allows.
+      
+    n_jobs: int, optional
+      For num_pixel_issues:
+      Number of processes for multiprocessing (default value = 1). Only used on Linux.
+      If `n_jobs=None`, will use either the number of: physical cores if psutil is installed, or logical cores otherwise.
+    
+    verbose : bool, optional
+      For num_pixel_issues:
+      Whether to suppress print statements or not.
+      
+    **kwargs:
+      downsample : int, optional
+      Factor to shrink labels and pred_probs by for 'num_pixel_issues' only . Default ``16``
+      Must be a factor divisible by both the labels and the pred_probs. Note that larger factors result in a linear 
+      decrease in performance
+
+
+
+    Returns
+    -------
+
+    image_scores: np.ndarray
+        Array of shape ``(N, )`` of scores between 0 and 1, one per image in the dataset.
+        Lower scores indicate image more likely to contain a label issue.
+    pixel_scores:
+        Array of shape ``(N,H,W)`` of scores between 0 and 1, one per pixel in the dataset.
+
+    """
+        
+    def check_input(labels: np.ndarray, pred_probs: np.ndarray) -> None:
```

Comment:
A lot of this code can be shared with `check_input()` in `filter`. Try and combine this into a single private function and use it for both.